src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ActiveConfiguredProjectsIgnoringTargetFrameworkProvider.cs

```diff
@@ -0,0 +1,180 @@
+        // Read/writes for both these fields must be done within a lock to keep them in sync.
+        private ImmutableDictionary<string, ConfiguredProject> _cachedConfiguredProjectsMap;
+        private string _cachedTargetFrameworks;
+
+        private IDisposable _evaluationSubscriptionLink;
+
+        [ImportingConstructor]
+        public ActiveConfiguredProjectsIgnoringTargetFrameworkProvider(
+            IUnconfiguredProjectServices services,
+            IUnconfiguredProjectCommonServices commonServices,
+            [Import(ExportContractNames.Scopes.UnconfiguredProject)]IProjectAsynchronousTasksService tasksService,
+            IActiveConfiguredProjectSubscriptionService activeConfiguredProjectSubscriptionService)
+            : base(commonServices.ThreadingService.JoinableTaskContext)
+        {
+            Requires.NotNull(services, nameof(services));
+            Requires.NotNull(commonServices, nameof(commonServices));
+            Requires.NotNull(tasksService, nameof(tasksService));
+            Requires.NotNull(activeConfiguredProjectSubscriptionService, nameof(activeConfiguredProjectSubscriptionService));
+
+            _services = services;
+            _commonServices = commonServices;
+            _tasksService = tasksService;
+            _activeConfiguredProjectSubscriptionService = activeConfiguredProjectSubscriptionService;
+        }
+
+        [ProjectAutoLoad(startAfter: ProjectLoadCheckpoint.ProjectFactoryCompleted)]
+        [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+        internal async Task OnProjectFactoryCompletedAsync()
+        {
+            await InitializeCoreAsync(CancellationToken.None).ConfigureAwait(false);
+        }
+
+        protected override async Task InitializeCoreAsync(CancellationToken cancellationToken)
+        {
+            using (_tasksService.LoadedProject())
+            {
+                await GetActiveConfiguredProjectsMapAsync().ConfigureAwait(false);
+
+                // Listen to changes to "TargetFrameworks" property.
+                var watchedEvaluationRules = Empty.OrdinalIgnoreCaseStringSet.Add(ConfigurationGeneral.SchemaName);
+                _evaluationSubscriptionLink = _activeConfiguredProjectSubscriptionService.ProjectRuleSource.SourceBlock.LinkTo(
+                        new ActionBlock<IProjectVersionedValue<IProjectSubscriptionUpdate>>(OnProjectChanged),
+                        ruleNames: watchedEvaluationRules, suppressVersionOnlyUpdates: true);
+            }
+        }
+
+        private async Task OnProjectChanged(IProjectVersionedValue<IProjectSubscriptionUpdate> e)
+        {
+            IProjectChangeDescription projectChange = e.Value.ProjectChanges[ConfigurationGeneral.SchemaName];
+            if (projectChange.Difference.ChangedProperties.Contains(ConfigurationGeneral.TargetFrameworksProperty))
+            {
+                var targetFrameworks = projectChange.After.Properties[ConfigurationGeneral.TargetFrameworksProperty];
```

Comment:
Clarification: these user-written forms are meaningful with respect to a given project, but not a globally unique way to describe a TFM. It depends how this data would be consumed. If it is to know what to pass to the project to instantiate itself as an inner build, you want the spelling in TargetFrameworks. If it's to reason about the resulting TFM, then you should instantiate the project with /p:TargetFramework={user spelling in Target Frameworks> and ask for $(TargetFrameworkMoniker).